### ISSUE TITLE
Set-up Redux with localStorage integration

### ISSUE THREAD

**Issue by joshmorel** [chingu-coders/voyage1-turtles-team-38#7]
We will use Redux for state management and persist some data (settings, to do lists, note taker) to localStorage.

Since we won't have any widgets created yet to use this, you can test just within the main page - display, update and persist a **note** object which will later be put into the note taker widget.

Consider testing with Jest: https://github.com/facebookincubator/create-react-app/blob/master/packages/react-scripts/template/README.md#running-tests

**Comment by joshmorel** [comment]
This library works nicely: https://github.com/elgerlambert/redux-localstorage

All we need in index.js to sync all state is in the code snipped that follows. If we want to sync only some of the state we need to use the `config.slicer`. So perhaps we don't want to store all the API data, but just settings, notes, todos. Something to consider.

```javascript
import persistState from 'redux-localstorage';
//...
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const enhancer = composeEnhancers(
  applyMiddleware(reduxThunk),
  persistState(null, {key: "leohClone"})
);
const store = createStore(reducers, {}, enhancer);
//...
```

**Comment by joshmorel** [comment]
Used redux-persist instead as more actively maintained - https://github.com/rt2zz/redux-persist

Resolved via #25
